tests/integration/test_interpreter_selection.py

```diff
@@ -397,109 +397,108 @@ def test_pex_python():
         assert py310 in stdout.decode("utf-8")
 
 
-def test_interpreter_selection_using_os_environ_for_bootstrap_reexec(pex_project_dir):
-    # type: (str) -> None
+def test_interpreter_selection_using_os_environ_for_bootstrap_reexec(
+    tmpdir,  # type: Any
+    pex_project_dir,  # type: str
+):
+    # type: (...) -> None
     """This is a test for verifying the proper function of the pex bootstrapper's interpreter
     selection logic and validate a corresponding bugfix.
 
     More details on the nature of the bug can be found at:
     https://github.com/pantsbuild/pex/pull/441
     """
-    with temporary_dir() as td:
-        pexrc_path = os.path.join(td, ".pexrc")
+    td = os.path.join(str(tmpdir), "tester_project")
+    pexrc_path = os.path.join(td, ".pexrc")
+
+    # Select pexrc interpreter versions based on test environment.
+    # The parent interpreter is the interpreter we expect the parent pex to
+    # execute with. The child interpreter is the interpreter we expect the
+    # child pex to execute with.
+    if sys.version_info[:2] == (3, 10):
+        child_pex_interpreter_version = PY39
+    else:
+        child_pex_interpreter_version = PY310
+
+    # Write parent pex's pexrc.
+    with safe_open(pexrc_path, "w") as pexrc:
+        pexrc.write("PEX_PYTHON={}".format(sys.executable))
+
+    test_setup_path = os.path.join(td, "setup.py")
+    with safe_open(test_setup_path, "w") as fh:
+        fh.write(
+            dedent(
+                """
+                from setuptools import setup
 
-        # Select pexrc interpreter versions based on test environment.
-        # The parent interpreter is the interpreter we expect the parent pex to
-        # execute with. The child interpreter is the interpreter we expect the
-        # child pex to execute with.
-        if sys.version_info[:2] == (3, 10):
-            child_pex_interpreter_version = PY310
-        else:
-            child_pex_interpreter_version = PY39
 
-        # Write parent pex's pexrc.
-        with open(pexrc_path, "w") as pexrc:
-            pexrc.write("PEX_PYTHON=%s" % sys.executable)
-
-        # The code below depends on pex.testing which depends on pytest - make sure the built pex
-        # gets this dep.
-        pytest_dist = find_distribution(ProjectName("pytest"))
-        assert pytest_dist is not None
-
-        test_setup_path = os.path.join(td, "setup.py")
-        with open(test_setup_path, "w") as fh:
-            fh.write(
-                dedent(
-                    """
-                    from setuptools import setup
-
-                    setup(
-                        name='tester',
-                        version='1.0',
-                        description='tests',
-                        author='tester',
-                        author_email='test@test.com',
-                        packages=['testing'],
```

Comment:
I indulged a bit here with more changes than needed. The issue that needed to be fixed was just re-naming `testing` to `tester` (The `testing` package is now a top-level package in the Pex repo and clashed leading to test failure).

While reading all this I noticed it was a bit of a horror show from the Twitter days. It still is but is a bit cleaner now.